### PR TITLE
[3.x] Fix duplicated edges in convex hull generator

### LINKED ISSUE #51909: Performance issue in GodotPhysics
### Godot version

3.4-beta4

### System information

OS: ubuntu 18.04

### Issue description

There is a performance issue in GodotPhysics that seems doesn't exist in v3.3.2-stable. Bullet is OK in both versions.

Godot v3.3.2-stable:

https://user-images.githubusercontent.com/5120436/130202793-30236392-5f72-4b3e-8e55-d08129d7de41.mp4

Godot v3.4-beta4:

https://user-images.githubusercontent.com/5120436/130203150-b64ebdd4-054a-4256-a338-609364c62b35.mp4

Bullet:

https://user-images.githubusercontent.com/5120436/130203774-4a44fa0e-8c6e-4053-b067-8f655aa2eb1b.mp4

### Steps to reproduce

Run the attached project.
If textures doesn't show, select QodotMap node, select "addons/qodot/example_scenes/1-interactivity/1-brush-entities/1-brush-entities.map" in Map File property and press "Full Build" button.

Screenshot from 2021-08-20 13-02-49

### Minimal reproduction project

[GodotPhysicsIssue.zip](https://github.com/godotengine/godot/files/7020265/GodotPhysicsIssue.zip)

### PR DESCRIPTION
**Pull request by pouleyKetchoupp**
All edges where accumulated from all faces, which ended up with double the amount in the final mesh data leading to performance issues in godot physics.

Fix #51909 (regression from #48533)
CC @mortarroad if you would like to review

If this fix is ok I'll make a PR on master tomorrow.

### CODE REVIEW

File: core/math/convex_hull.cpp [hunk 1]
```diff
@@ -2257,22 +2258,21 @@ Error ConvexHullComputer::convex_hull(const PoolVector<Vector3> &p_points, Geome
...
 	if (p_point_count == 0) {
-		return FAILED; // matches QuickHull
+		return FAILED; // Matches QuickHull.
 	}
 
 	ConvexHullComputer ch;
 	ch.compute(p_points, p_point_count, -1.0, -1.0);
 
 	r_mesh.vertices = ch.vertices;
+	int vertex_count = r_mesh.vertices.size();
```
Comment (lawnjelly) [inline core/math/convex_hull.cpp:2271]: Could do with a comment here and at the resize at the bottom of the function to explain why this is being done - it is clear in the PR text, but not in the source.

File: core/math/convex_hull.cpp [hunk 2]
```diff
@@ -2283,25 +2283,39 @@ Error ConvexHullComputer::convex_hull(const Vector3 *p_points, int32_t p_point_c
...
+		// Compute normal.
 		if (face.indices.size() >= 3) {
 			face.plane = Plane(r_mesh.vertices[face.indices[0]], r_mesh.vertices[face.indices[1]], r_mesh.vertices[face.indices[2]]);
 		} else {
 			WARN_PRINT("Too few vertices per face.");
 		}
 	}
 
+	r_mesh.edges.resize(edge_count);
+
```
Comment (lawnjelly) [inline core/math/convex_hull.cpp:2319]: Add comment to explain.

**Comment by lawnjelly** [comment]
Looks fine to me, I'd just add a couple of comments to explain what is going on in the source because it isn't immediately clear.

**Comment by mortarroad** [comment]
Ah, you're of course right. Since this is a half edge data structure, there's a *reverse* edge for every edge.

Rather than going through a set, you can simply pick one of the two as the canonical edge. (E.g. the one where `source_vertex < target_vertex`, which will only be true for one of them.)

**Comment by mortarroad** [comment]
I made a PR with my approach: https://github.com/godotengine/godot/pull/52059

**Comment by pouleyKetchoupp** [comment]
Closing in favor of #52059.

**Pull request closed without merging**